backend/src/services/event_service.py

```
@@ -311,6 +311,185 @@ def get_stats(self, team_id: int) -> dict:
+
+        if preset == "upcoming_30d":
+            # Only upcoming_30d has the 30-day window
+            end_date = today + timedelta(days=30)
+            query = query.filter(Event.event_date <= end_date)
+        elif preset == "needs_tickets":
+            effective_ticket_required = case(
+                (Event.ticket_required.isnot(None), Event.ticket_required),
+                else_=EventSeries.ticket_required,
+            )
+            query = query.filter(
+                effective_ticket_required == True,
```

Ruff (0.15.0)

[error] Avoid equality comparisons to `True`; use `effective_ticket_required:` for truth checks

Replace with `effective_ticket_required`

(E712)